test/integration/tests/modules.cpp

```diff
@@ -480,5 +480,29 @@ TEST_F(ZNCTest, SaslAuthExternal) {
         ":irc.znc.in 904 nick :SASL authentication failed");
 }
 
+TEST_F(ZNCTest, AutoMode) {
+    auto znc = Run();
+    auto ircd = ConnectIRCd();
+    auto client = LoginClient();
+    client.Write("znc loadmod automode");
```

Comment:
This only does the UI, but doesn't check the functionality of automatically giving voice/ops with/without keys

Reply:
Tests with keys have been written and work from my testing.